clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/helpers/AbstractServerGroupNameResolver.groovy

```diff
@@ -72,10 +86,14 @@ abstract class AbstractServerGroupNameResolver extends NameBuilder {
     if (stepCounter == SEQUENTIAL_NUMBERING_NAMESPACE_SIZE) {
       throw new IllegalArgumentException("All server group names for cluster $clusterName in $region are taken.")
     }
-
     return generateServerGroupName(application, stack, details, nextSequence, ignoreSequence)
   }
 
+  protected int generateNextSequence(String serverGroupName) {
+    Names parts = Names.parseName(serverGroupName)
+    return ((parts.sequence ?: 0) + 1) % SEQUENTIAL_NUMBERING_NAMESPACE_SIZE
```

Comment:
Maybe have this line in just one helper method to be called here and on line 77?